lib/util/location.ex

```diff
@@ -115,39 +196,16 @@ defmodule Util.Location do
         ) :: __MODULE__.t()
 
   def nearest_point_to_segment(
-        %__MODULE__{latitude: latitude, longitude: longitude},
-        {%{latitude: start_latitude, longitude: start_longitude},
-         %{latitude: end_latitude, longitude: end_longitude}}
+        point = %__MODULE__{},
+        {start_loc = %__MODULE__{}, end_loc = %__MODULE__{}}
       ) do
-    # Scale factors used to transform between lat/long and a coordinate
-    # system based on distance
-    latitude_scale_factor =
-      1000 *
-        distance(
-          new(start_latitude, start_longitude),
-          new(start_latitude + 0.001, start_longitude)
-        )
-
-    longitude_scale_factor =
-      1000 *
-        distance(
-          new(start_latitude, start_longitude),
-          new(start_latitude, start_longitude + 0.001)
-        )
-
-    # Use the above scale factors to transform both the original
-    # segment and the point in question to (x, y) coordinates, where
-    # the origin is the segment start point, x is meters east from
-    # there, and y is meters north
-    segment_end_x = longitude_scale_factor * (end_longitude - start_longitude)
-    segment_end_y = latitude_scale_factor * (end_latitude - start_latitude)
+    segment_vector = displacement(start_loc, end_loc)
 
-    x = longitude_scale_factor * (longitude - start_longitude)
-    y = latitude_scale_factor * (latitude - start_latitude)
+    point_vector = displacement(start_loc, point)
```

Comment:
`point_vector` seems like a silly name, but I'm not sure of a better one. It's the vector pointing from the `start` point of the segment to the non-segment point.